src/vivarium_public_health/metrics/disability.py

```diff
@@ -150,6 +152,12 @@ def _register_metrics_modifier(self, builder: Builder) -> None:
     # Event-driven methods #
     ########################
 
+    # noinspection PyUnusedLocal
+    def on_post_setup(self, event: Event) -> None:
```

Comment:
for my own understanding: is this post-setup step new / is its sole purpose to check if the cause is applicable?

Reply:
This is because we don't have guarantees on the order in which components have their setup methods run. The disability and mortality components need all components that potentially have excess disability weight or excess mortality to have run their setup so that the `has_disability_weight` or `has_excess_mortality` fields have been created on them.